Move patch invalidateCaps in Entity

ferriarnus opened the pull request:
Move misaligned patch in Entity from processFlappingMovement to setRemoved. 
fixes #8690

ferriarnus committed:
Move patch invalidateCaps in Entity

Move misaligned patch in Entity from processFlappingMovement to setRemoved.

sciwhiz12 reviewed (changes requested):
Looking at the 1.18.x patch, the `invalidateCaps` method was patched into `Entity#remove`, so this fix should follow that (unless there is a reason for moving it into `Entity#setRemoved` rather then `#remove`).

ferriarnus committed:
Update Entity.java.patch

Apologies, it seems I made a mistake. The patch in 1.18 is indeed on "remove", and as it appears nothing changed with it's calls, should stay there. The only difference would be that caps are not invalidated on dimension change, which is intended.

ferriarnus commented:
Apologies, it seems I made a mistake. The patch in 1.18 is indeed on "remove", and as it appears nothing changed with it's calls, should stay there.

sciwhiz12 reviewed (approved):
Please add `fixes #8690` to your PR description, so GitHub automatically links the issue to this PR.
